test/basic_test.rb

```diff
@@ -61,4 +61,18 @@ class BasicTest < ActiveSupport::TestCase
     second_number = archived.archive_number
     assert_equal initial_number, second_number
   end
+
+  test "archived=true archives the record" do
+    archival = Archival.create!
+    archival.archived = true
+    archival.save!
```

Comment:
The `#save!` call here and below is unnecessary, as `#archive` and `#unarchive` already perform the `#save!` call as part of the transaction.